src/main/webapp/js/actions/auth-actions.js

```diff
@@ -25,6 +29,46 @@ export const authError = error => ({
   payload: error,
 });
 
+export const signInOffline = ({ password, savedLogin }, callback) => (dispatch) => {
+  if (savedLogin) {
+    checkHash(password, _.get(savedLogin, 'hash', ''), (result) => {
+      if (result) {
+        const tokenDecoded = jwtDecode(savedLogin.token);
+        dispatch({ type: AUTH_USER });
+        dispatch({
+          type: SET_PERMISSIONS,
+          payload: _.get(tokenDecoded, 'authorities', ''),
+        });
+        dispatch({
+          type: SET_COUNTER_LOGOUT_TIME,
+          payload: _.get(tokenDecoded, 'token.exp_period', ''),
+        });
+        localStorage.setItem('refresh_token', savedLogin.refreshToken);
+        localStorage.setItem('token', savedLogin.token);
+        callback();
+      } else {
+        dispatch(authError('Wrong username or password. Please try again.'));
+      }
+    });
+  } else {
+    dispatch(authError('Please try again in online mode.'));
+  }
+};
+
+export const storeLogin = (dispatch, username, password, token, refreshToken) => {
```

Comment:
move all of this to the signinUser action. also combine AUTH_USER, SET_COUNTER_LOGOUT_TIME, SET_PERMISSIONS and STORE_LOGIN into one action type